src/components/data/FileDataStorage.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+namespace yii\debug\components\data;
+
+use Yii;
+use yii\base\Component;
+use yii\base\InvalidConfigException;
+use yii\debug\FlattenException;
+use yii\debug\Module;
+use yii\helpers\FileHelper;
+
+class FileDataStorage extends Component implements DataStorage
+{
+    /**
+     * @var string
+     */
+    public $dataPath = '@runtime/debug';
+
+    /**
+     * @var int the permission to be set for newly created directories.
+     * This value will be used by PHP [[chmod()]] function. No umask will be applied.
+     * Defaults to 0775, meaning the directory is read-writable by owner and group,
+     * but read-only for other users.
+     * @since 2.0.6
+     */
+    public $dirMode = 0775;
+
+    /**
+     * @var int the permission to be set for newly created debugger data files.
+     * This value will be used by PHP [[chmod()]] function. No umask will be applied.
+     * If not set, the permission will be determined by the current environment.
+     * @since 2.0.6
+     */
+    public $fileMode;
+
+    /**
+     * @var Module
+     */
+    private $module;
+
+    /**
+     * @var int the maximum number of debug data files to keep. If there are more files generated,
+     * the oldest ones will be removed.
+     */
+    public $historySize = 50;
+
+    /**
+     * @var array
+     */
+    private $_manifest;
+
+    /**
+     * @return void
+     */
+    public function init()
+    {
+        parent::init();
+        $this->dataPath = Yii::getAlias($this->dataPath);
```

Comment:
We need proper initialization here as well.

Reply:
what do you mean?

Reply:
See [here](https://github.com/yiisoft/yii2/blob/master/framework/rbac/DbManager.php#L119) for example.

Reply:
But Instance::ensure is not appropriate here, as well as checking for null.

This code has been moved here from this [place](https://github.com/yiisoft/yii2-debug/blob/master/src/Module.php#L223)

Reply:
Oh, yes, sorry, this is not a component. My bad.